frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -120,6 +131,29 @@ public int getNumRunningWorkers(String modelName) {
         }
     }
 
+    public void addServerThread(Model model, CompletableFuture<HttpResponseStatus> future)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        WorkerStateListener listener = new WorkerStateListener(future, 1);
+        BatchAggregator aggregator = new BatchAggregator(model);
+        synchronized (model.getModelName()) {
+            model.setPort(port.getAndIncrement());
+            WorkerThread thread =
+                    new WorkerThread(
+                            configManager,
+                            backendGroup,
+                            model.getPort(),
+                            -1,
```

Comment:
The GPU ID -1 would mean its CPU. This means, server thread is always launched on CPUs. Just for sanity, test this code on GPU hosts as well.